go.mod: bump redact package version to v1.1.6

arjunmahishi opened the pull request:
This version introduces two new APIs: `SafeByte`, `SafeBytes`. These are useful for the upcoming changes in #138632.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/140095)
<!-- Reviewable:end -->

arjunmahishi committed:
go.mod: bump redact package version to v1.1.6

This version introduces two new APIs: SafeByte, SafeBytes. These are
useful for the upcoming changes in #138632.

Epic: none
Release note: None

arjunmahishi commented:
The failing test is not related to this change. Looks like it was inherited from master.

arjunmahishi commented:
TFTR!
bors r=dhartunian
